web-services/client/src/main/java/datawave/webservice/model/ModelValidator.java

```diff
@@ -0,0 +1,119 @@
+package datawave.webservice.model;
+
+import datawave.query.util.MetadataHelper;
+import datawave.util.UniversalSet;
+import org.apache.accumulo.core.client.TableNotFoundException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+public class ModelValidator {
+    
+    public static void validate(MetadataHelper metadataHelper, Model model) throws InvalidModelException, TableNotFoundException {
+        
+        Map<String,Set<String>> forwardMappings = buildForwardMappingMap(model);
+        checkFieldProperties(metadataHelper, forwardMappings);
+        // anything else to check?
```

Comment:
I seem to recall a case where two separate fields may share a reverse mapping to the same field. In this case when the reverse mapping is applied, it's difficult to determine which query field actually hit. Maybe a log a warning if more than one field contributes to a single reverse mapped field.

```
MODEL_FIELD_A ==> DB_FIELD_1, FWD
MODEL_FIELD_B ==> DB_FIELD_2, FWD

MODEL_FIELD_A ==> DB_FIELD_1, REV
MODEL_FIELD_A ==> DB_FIELD_1, REV
```

It may also be worth checking verifying there are no duplicates between multiple database fields reverse mapping to the same model field.

Reply:
I like it.  Thanks!